### ISSUE TITLE
Feature request: is it possible to sync the content from trakt to Plex even tho the content isn't available on plexserver

### ISSUE THREAD

**Issue by krishne35** [Taxel/PlexTraktSync#1142]
### Use case

Basically I use plex_debrid service which scrapes content and skips if it's already watched so is it possible to copy the watched status of shows from trak to plex account even if it's show isn't available on server
### Workarounds

Manually marking content as watched on plex

**Comment by RileyXX** [comment]
Reposting my comment from [another post](https://github.com/Taxel/PlexTraktSync/discussions/1293) here:

So I know this script can import watch history from Trakt into Plex and vise versa. But from what I can tell it only imports watch history for items that are currently available to watch in my Plex library. The thing is I want to add ALL my watch history for items dating back several years into my Plex library, including items that are no longer available to watch in my Plex library. That way when I add old content to my Plex library the watch history is already available. Is this possible with this script?

I currently have the script set to auto run and import my watch history from plex to trakt. But I have trakt to plex disabled because sometimes things get marked as played by accident. And I have all my shows set to auto delete watched episodes. So in my case it's not worth it to auto import from trakt to plex since things will randomly get deleted on occasion. So I'm trying to find a way to import my old watch history all at once that way I don't need to have trakt to plex sync enabled.

**Comment by glensc** [comment]
I don't understand how can you mark an item watch state in plex that doesn't exist in plex? you need to have plex item to record state somewhere. the media files can of course be missing.

**Comment by twolaw** [comment]
I self-assigned this.
@glensc you are misunderstanding the feature request. Let me work it. I know how to do.

**Comment by twolaw** [comment]
Needs to be implemented in PlexAPI first :
- https://github.com/pkkid/python-plexapi/issues/1067

**Comment by RileyXX** [comment]
Not sure if this would be helpful but there is another script for importing/exporting plex data. I'm not sure if this script marks items as played that are not available in Plex but I'll just post it here maybe it will be useful as a point of reference. I don't think this script uses the PlexAPI. https://github.com/Casvt/Plex-scripts/blob/main/multiple_servers/plex_exporter_importer.py

**Comment by twolaw** [comment]
No, we'll do it properly with Plex API.

**Comment by glensc** [comment]
> > No, we'll do it properly with Plex API.
> 
> Fair enough. In the meantime if it's not possible with plex api I think I'll try putting together my own little script to see if I can accomplish this

Why not put your efforts to the right place and submit a fix/improvement to PlexAPI?

**Comment by RileyXX** [comment]
> > > No, we'll do it properly with Plex API.
> > 
> > 
> > Fair enough. In the meantime if it's not possible with plex api I think I'll try putting together my own little script to see if I can accomplish this
> 
> Why not put your efforts to the right place and submit a fix/improvement to PlexAPI?

Yeah you're right. I can take a crack at it. But I am a bit of an amateur python coder. I bet [@Casvt](https://github.com/Casvt) could figure this out.

**Comment by twolaw** [comment]
I'm currently working on it.

To sync :

watched status of :
- movies
- episodes

and ratings of :
- movies
- shows
- seasons
- episodes

both ways between Trakt and Plex items _not in the library_.

The user is then supposed to enable the [Sync Watch states and Ratings](https://support.plex.tv/articles/sync-watch-state-and-ratings/) to use it at full potential.

Side effect : when disabling a trakt watched status or rating, it will be back at next sync because Plex keeps it in cloud database. Users may think it's a bug.

**Comment by RileyXX** [comment]
> I'm currently working on it.

Very cool thanks. Looking forward to seeing this in an official release.

**Comment by glensc** [comment]
@twolaw if you have unfinished work on this. perhaps push to some PR. a bit easier to continue one's work than start from scratch.

**Comment by twolaw** [comment]
I need to find time to continue this.
Here is bulk commit from May-3 : https://github.com/twolaw/PlexTraktSync/tree/plex-cloud-watched-status

**Comment by glensc** [comment]
@simonc56 please create a draft PR then.

**Comment by simonc56** [comment]
Draft PR is here :
-  #1676

**Comment by tomsinclair94** [comment]
Hi guys,

Didn't want to open another Issue so searched around first. I've got everything working with the sync tool, but the one thing 'missing' for me is the ability to sync watched progress on non-Plex media in my watch list. Anything on my watch list that is in my Plex local library syncs perfectly, but anything not in my library does not sync any watched status from Trakt.

I stumbled across a few issues reported that landed me here. I can see there is a Draft PR open and I just wanted to check-in to make sure I'm not doing something wrong and the feature is available _**or**_ it's just not completed and in place as of yet.

**Comment by glensc** [comment]
Nobody is working on the feature.

The pr you found is stale, nobody who was interested of it, gave their feedback there (only code-review notes there), or feedback here, so apparently nobody even tried the pr.

**Comment by BLACK4585** [comment]
Just leaving my comment here to show my Interest in this feature.
